Q1- A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.



In [1]:
import pandas as pd 
import numpy as np

df=pd.read_csv('cutlets.csv')
df.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [6]:
unitA=df['Unit A']
unitB=df['Unit B']

In [7]:
from scipy import stats                  # this is ANOVA test by this we have calculated p-value and decide to null hypothesis
stats.f_oneway(unitA, unitB)  

F_onewayResult(statistic=0.5225394038913932, pvalue=0.47223947245995734)

*  according given significance level 5% p-vlaue is low so null hypothesis is rejected, So values 
 of diameter both data is different

****************************************

Q-2 A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
   
  Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.
 
 
    Minitab File: LabTAT.mtw


In [1]:
import pandas as pd
import numpy as np
df =pd.read_csv('LabTAT.csv')
df.head()


Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [2]:
Laboratory1= df['Laboratory 1']
Laboratory2= df['Laboratory 2']
Laboratory3= df['Laboratory 3']
Laboratory4= df['Laboratory 4']

In [3]:
Laboratory1.mean(), Laboratory2.mean(), Laboratory3.mean(), Laboratory4.mean()

(178.36158333333336, 178.9029166666667, 199.91324999999998, 163.68275)

In [4]:
from scipy import stats
stats.f_oneway(Laboratory1, Laboratory2, Laboratory3, Laboratory4) 

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

* Significane level = 5% so according this vale p-value very low so null hypothesis is rejected its mean time of library are 
diffrent one to each other.

**********************************************

Q-3   Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.


In [6]:
import pandas as pd
import numpy as np
df =pd.read_csv('BuyerRatio.csv')
df

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [7]:
df1=df.drop(['Observed Values'], axis=1)
df1

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [8]:
df2= pd.DataFrame({'Males':[50,142,131,70], 'Females':[435,1523,1356,750]})
df2

Males  Females
0     50      435
1    142     1523
2    131     1356
3     70      750

In [9]:
males1 = df2['Males']
females1= df2['Females']

In [10]:
from scipy import stats
stats.f_oneway(males1,females1)

F_onewayResult(statistic=12.841384359570121, pvalue=0.011594334372357063)

* here if consider sigificence value 5% = 0.05 so null hypothesis is rejected its mean buying by product males and females are 
diffrent 

*************************

Q-4 TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

Minitab File: CustomerOrderForm.mtw
 


In [11]:
import pandas as pd 
import numpy as np 
df=pd.read_csv('Costomer+OrderForm.csv')
df

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [12]:
df['Phillippines'].value_counts(), df['Indonesia'].value_counts(), df['Malta'].value_counts(), df['India'].value_counts()

(Error Free    271
 Defective      29
 Name: Phillippines, dtype: int64,
 Error Free    267
 Defective      33
 Name: Indonesia, dtype: int64,
 Error Free    269
 Defective      31
 Name: Malta, dtype: int64,
 Error Free    280
 Defective      20
 Name: India, dtype: int64)

In [13]:
from scipy.stats import chi2_contingency

data= [[271,267,269,280],[29,33,31,20]]   # here I have used Chi-square test 
chi2_contingency(data)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

* here null hypothesis is accepted because significane value is 0.05 and p value is 0.2771 that is high
* its mean deffective is varying by center